File: services/web/src/i18n/labels.ts
/**
 * Человеческие подписи для перечислений gamification.
 *
 * Вынесены отдельно, потому что нужны сразу трём экранам (`InstitutionsPage`,
 * `InvitationsPage`, `InvitePage`) — дублировать карты по месту не стоило.
 */
import type { InstitutionKind, MembershipStatus, UserRole } from '../api/auth'
import type { TransactionKind } from '../api/currency'
import type { PurchaseStatus } from '../api/market'

export const KIND_LABELS: Record<InstitutionKind, string> = {
  school: 'школа',
  camp: 'лагерь',
}

export const TRANSACTION_KIND_LABELS: Record<TransactionKind, string> = {
  manual_accrual: 'Начисление',
  reversal: 'Сторно',
  purchase: 'Покупка',
  purchase_refund: 'Возврат за покупку',
}

export const PURCHASE_STATUS_LABELS: Record<PurchaseStatus, string> = {
  pending: 'Ожидает решения',
  fulfilled: 'Выдано',
  rejected: 'Отклонено',
}

export const ROLE_LABELS: Record<UserRole, string> = {
  student: 'ученик',
  teacher: 'преподаватель',
  institution_admin: 'администратор',
}

export const STATUS_LABELS: Record<MembershipStatus, string> = {
  invited: 'приглашение не принято',
  active: 'доступ активен',
  suspended: 'доступ приостановлен',
}

File: services/mobile/lib/i18n/labels.dart
/// Человеческие подписи для перечислений — зеркало
/// `services/web/src/i18n/labels.ts`. Только то, что нужно этапу 09a:
/// подписи ролей, статусов членства, типов учреждения и операций валюты
/// (перечисления `institution_admin`/каталога — 09b/09c — не переносятся,
/// пока их модели не появились в `lib/api/`).
library;

import '../api/auth_api.dart';
import '../api/currency_api.dart';
import '../api/market_api.dart';

const Map<InstitutionKind, String> kindLabels = {
  InstitutionKind.school: 'школа',
  InstitutionKind.camp: 'лагерь',
};

const Map<TransactionKind, String> transactionKindLabels = {
  TransactionKind.manualAccrual: 'Начисление',
  TransactionKind.reversal: 'Сторно',
  TransactionKind.purchase: 'Покупка',
  TransactionKind.purchaseRefund: 'Возврат за покупку',
};

const Map<PurchaseStatus, String> purchaseStatusLabels = {
  PurchaseStatus.pending: 'Ожидает решения',
  PurchaseStatus.fulfilled: 'Выдано',
  PurchaseStatus.rejected: 'Отклонено',
};

const Map<UserRole, String> roleLabels = {
  UserRole.student: 'ученик',
  UserRole.teacher: 'преподаватель',
  UserRole.institutionAdmin: 'администратор',
};

const Map<MembershipStatus, String> statusLabels = {
  MembershipStatus.invited: 'приглашение не принято',
  MembershipStatus.active: 'доступ активен',
  MembershipStatus.suspended: 'доступ приостановлен',
};
